.github/copilot-instructions.md

```diff
@@ -0,0 +1,194 @@
+# Copilot Instructions for gradle-common
+
+## Repository Overview
+
+**Huanshankeji Gradle Common** is a collection of Gradle plugins and common dependencies for Kotlin projects. The repository provides convention plugins, dependency management, and build utilities primarily used by Huanshankeji for their Kotlin projects.
+
+**High-level Repository Information:**
+- **Project Type:** Gradle plugin library with multiple modules
+- **Languages:** Kotlin (primary), Gradle Kotlin DSL
+- **Target Runtime:** JVM (JDK 8+, tested with JDK 8 and 11)
+- **Framework:** Gradle 9.0.0, Kotlin 2.2.20
+- **Size:** Medium-sized multi-module project (~20 plugin modules)
+- **APIs:** Experimental and subject to change
+
+## Build Instructions
+
+### Prerequisites
+- JDK 8 or higher (JDK 8 is used in CI, JDK 11 is available)
+- Gradle wrapper handles Gradle version automatically
+
+### Essential Build Commands
+
+**Always run commands in the repository root directory.**
+
+#### Bootstrap and Dependencies
+```bash
+# No special bootstrap needed - Gradle wrapper handles dependencies
+./gradlew --version  # Verify Gradle 9.0.0 is used
+```
+
+#### Build
+```bash
+./gradlew build     # Full build (takes ~30 seconds, includes tests)
+./gradlew check     # Run checks without building distributions (~1 minute)
+```
+
+#### Test
+```bash
+./gradlew test      # Run all tests (minimal test suite)
+```
+
+#### Clean and Rebuild
+```bash
+./gradlew clean     # Clean all build outputs
+./gradlew build     # Rebuild from scratch
+```
+
+#### Local Publishing
+```bash
+./gradlew publishToMavenLocal  # Publish to local Maven repo (~2 seconds)
+```
+
+#### Documentation Generation
+```bash
+./gradlew :dokkaGeneratePublicationHtml  # Generate API docs with Dokka
+./gradlew dokkaGenerate                  # Alternative Dokka generation command
+```
+
+#### API Validation
+```bash
+./gradlew apiCheck    # Check API compatibility against golden files
+./gradlew apiDump     # Update API golden files for modules
+./gradlew validatePlugins  # Validate plugin parameter annotations
+```
+
+### Known Build Issues and Workarounds
+
+**Warning Messages (Safe to Ignore):**
+- "Unsupported Kotlin plugin version" warnings related to Gradle Kotlin DSL compatibility
+- Dokka V2 migration messages
```

Comment:
This should no longer appear after Dokka is bumped to 2.1.0-Beta. Check again and remove if so.